Frameworks/UIKit/UIViewController.mm

```diff
@@ -1139,9 +1144,7 @@ - (void)presentViewController:(UIViewController*)controller animated:(BOOL)anima
         return;
     }
 
-    UIViewController* oldViewController = self;
-    if (priv->_modalViewController != nil) {
-        oldViewController = priv->_modalViewController;
+    if ([self presentedViewController]) {
```

Comment:
I was thinking for cases where the property just returns the ivar (at least in UIViewController) it's better to access via the property for the benefit of subclassing (not that subclassing presentedViewController is likely to work or be useful).